### ISSUE TITLE
Rule: require suggestions have different `messageId` than their parent report

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint-community/eslint-plugin-eslint-plugin#368]
Code like this happens once in a while from contributors unfamiliar with ESLint rule suggestions:

```js
context.report({
  node,
  messageId: "someMessage",
  suggest: [
    {
      messageId: "someMessage",
      // ...
    },
  ],
});
```

At least in typescript-eslint we try to use different messages for suggestions. Suggestion messages are typically action-oriented, whereas parent report messages are statements of fact.

What do you think about a lint rule helping devs know to use a different message between the two?

**Comment by bmish** [comment]
Sounds like a good idea. Perhaps a name like `no-matching-violation-suggest-message-ids`.
